src/main/groovy/org/jenkinsci/plugins/pipeline/config/parser/JSONParser.groovy

```diff
@@ -223,11 +223,15 @@ class JSONParser {
     }
 
     public @CheckForNull ConfigASTValue parseValue(JSONObject o) {
-        if (o.getBoolean("isConstant")) {
-            return ConfigASTValue.fromConstant(o.get("value"), o)
-        } else {
-            return ConfigASTValue.fromGString(o.getString("value"), o)
+        // TODO: Probably need better type-checking? Not sure yet.
+        if (o.getString("value") != null) {
+            String valString = o.getString("value")
+            if (valString.contains('$class')) {
```

Comment:
why is it looking for $class? Is this trying to discern between: 

" foo is ${bar}"
and 'foo is $bar' if it was regular groovy?